--- 2.12.15/scala/collection/immutable/Set.scala
+++ old-collections/scala/collection/immutable/Set.scala
@@ -63,13 +63,14 @@
  *  @define coll immutable set
  */
 object Set extends ImmutableSetFactory[Set] {
-  override def newBuilder[A]: mutable.Builder[A, Set[A]] = new SetBuilderImpl[A]
+  @inline override def newBuilder[A]: mutable.Builder[A, Set[A]] = new SetBuilderImpl[A]
 
   /** $setCanBuildFromInfo */
-  implicit def canBuildFrom[A]: CanBuildFrom[Coll, A, Set[A]] =
+  @inline implicit def canBuildFrom[A]: CanBuildFrom[Coll, A, Set[A]] =
     ReusableCBF.asInstanceOf[CanBuildFrom[Coll, A, Set[A]]]
   private[this] val ReusableCBF = setCanBuildFrom[Any]
 
+  @inline override def empty[A]: Set[A] = EmptySet.asInstanceOf[Set[A]]
   /** An optimized representation for immutable empty sets */
   @SerialVersionUID(-2443710944435909512L)
   private object EmptySet extends AbstractSet[Any] with Set[Any] with Serializable {
