--- 2.12.14/scala/collection/immutable/Set.scala
+++ old-collections/scala/collection/immutable/Set.scala
@@ -66,1 +66,1 @@
-  override def newBuilder[A]: mutable.Builder[A, Set[A]] = new SetBuilderImpl[A]
+  @inline override def newBuilder[A]: mutable.Builder[A, Set[A]] = new SetBuilderImpl[A]
@@ -69,1 +69,1 @@
-  implicit def canBuildFrom[A]: CanBuildFrom[Coll, A, Set[A]] =
+  @inline implicit def canBuildFrom[A]: CanBuildFrom[Coll, A, Set[A]] =
@@ -73,0 +73,2 @@
+  @inline override def empty[A]: Set[A] = EmptySet.asInstanceOf[Set[A]]
+
