--- 2.12.14/scala/collection/immutable/Stream.scala
+++ old-collections/scala/collection/immutable/Stream.scala
@@ -1079,0 +1079,4 @@
+  @inline override def ReusableCBF: GenericCanBuildFrom[Nothing] = ReusableCBFInstance
+  private object ReusableCBFInstance extends GenericCanBuildFrom[Nothing] {
+    @inline override def apply() = newBuilder[Nothing]
+  }
@@ -1090,3 +1092,1 @@
-  implicit def canBuildFrom[A]: CanBuildFrom[Coll, A, Stream[A]] =
-    ReusableCBF.asInstanceOf[CanBuildFrom[Coll, A, Stream[A]]]
-  private[this] val ReusableCBF = new StreamCanBuildFrom[Any]
+  @inline implicit def canBuildFrom[A]: CanBuildFrom[Coll, A, Stream[A]] = new StreamCanBuildFrom[A]
@@ -1095,1 +1095,1 @@
-  def newBuilder[A]: Builder[A, Stream[A]] = new StreamBuilder[A]
+  @inline def newBuilder[A]: Builder[A, Stream[A]] = new StreamBuilder[A]
