--- scalalib/target/scalaSources/2.12.13/scala/collection/generic/GenTraversableFactory.scala	2021-01-12 14:37:36.000000000 +0100
+++ scalalib/overrides-2.12/scala/collection/generic/GenTraversableFactory.scala	2021-02-19 18:54:20.722914699 +0100
@@ -43,6 +43,7 @@
 
-  private[this] val ReusableCBFInstance: GenericCanBuildFrom[Nothing] = new GenericCanBuildFrom[Nothing] {
-    override def apply() = newBuilder[Nothing]
+  private object ReusableCBFInstance extends GenericCanBuildFrom[Nothing] {
+    @inline override def apply() = newBuilder[Nothing]
   }
-  def ReusableCBF: GenericCanBuildFrom[Nothing] = ReusableCBFInstance
+
+  @inline def ReusableCBF: GenericCanBuildFrom[Nothing] = ReusableCBFInstance
 
