--- 2.12.14/scala/collection/mutable/ArrayStack.scala
+++ old-collections/scala/collection/mutable/ArrayStack.scala
@@ -27,3 +27,9 @@
-  implicit def canBuildFrom[A]: CanBuildFrom[Coll, A, ArrayStack[A]] = ReusableCBF.asInstanceOf[GenericCanBuildFrom[A]]
-  def newBuilder[A]: Builder[A, ArrayStack[A]] = new ArrayStack[A]
-  def empty: ArrayStack[Nothing] = new ArrayStack()
+  @inline override def ReusableCBF: GenericCanBuildFrom[Nothing] = ReusableCBFInstance
+  private object ReusableCBFInstance extends GenericCanBuildFrom[Nothing] {
+    @inline override def apply() = newBuilder[Nothing]
+  }
+
+  @inline implicit def canBuildFrom[A]: CanBuildFrom[Coll, A, ArrayStack[A]] = ReusableCBF.asInstanceOf[GenericCanBuildFrom[A]]
+  @inline def newBuilder[A]: Builder[A, ArrayStack[A]] = new ArrayStack[A]
+
+  @inline def empty: ArrayStack[Nothing] = new ArrayStack()
