--- 2.12.15/scala/collection/mutable/Buffer.scala
+++ old-collections/scala/collection/mutable/Buffer.scala
@@ -42,8 +42,12 @@
  *  @define Coll `Buffer`
  */
 object Buffer extends SeqFactory[Buffer] {
-  implicit def canBuildFrom[A]: CanBuildFrom[Coll, A, Buffer[A]] = ReusableCBF.asInstanceOf[GenericCanBuildFrom[A]]
-  def newBuilder[A]: Builder[A, Buffer[A]] = new ArrayBuffer
+  @inline override def ReusableCBF: GenericCanBuildFrom[Nothing] = ReusableCBFInstance
+  private object ReusableCBFInstance extends GenericCanBuildFrom[Nothing] {
+    @inline override def apply() = newBuilder[Nothing]
+  }
+  @inline implicit def canBuildFrom[A]: CanBuildFrom[Coll, A, Buffer[A]] = ReusableCBF.asInstanceOf[GenericCanBuildFrom[A]]
+  @inline def newBuilder[A]: Builder[A, Buffer[A]] = new ArrayBuffer
 }
 
 /** Explicit instantiation of the `Buffer` trait to reduce class file size in subclasses. */
