--- 2.12.14/scala/collection/mutable/Buffer.scala
+++ old-collections/scala/collection/mutable/Buffer.scala
@@ -45,3 +45,4 @@
-  implicit def canBuildFrom[A]: CanBuildFrom[Coll, A, Buffer[A]] = ReusableCBF.asInstanceOf[GenericCanBuildFrom[A]]
-  def newBuilder[A]: Builder[A, Buffer[A]] = new ArrayBuffer
-}
+  @inline override def ReusableCBF: GenericCanBuildFrom[Nothing] = ReusableCBFInstance
+  private object ReusableCBFInstance extends GenericCanBuildFrom[Nothing] {
+    @inline override def apply() = newBuilder[Nothing]
+  }
@@ -49,0 +50,3 @@
+  @inline implicit def canBuildFrom[A]: CanBuildFrom[Coll, A, Buffer[A]] = ReusableCBF.asInstanceOf[GenericCanBuildFrom[A]]
+  @inline def newBuilder[A]: Builder[A, Buffer[A]] = new ArrayBuffer
+}
