--- 2.12.14/scala/collection/mutable/DoubleLinkedList.scala
+++ old-collections/scala/collection/mutable/DoubleLinkedList.scala
@@ -85,1 +85,1 @@
-  implicit def canBuildFrom[A]: CanBuildFrom[Coll, A, DoubleLinkedList[A]] = ReusableCBF.asInstanceOf[GenericCanBuildFrom[A]]
+  @inline implicit def canBuildFrom[A]: CanBuildFrom[Coll, A, DoubleLinkedList[A]] = ReusableCBF.asInstanceOf[GenericCanBuildFrom[A]]
@@ -87,1 +87,1 @@
-  def newBuilder[A]: Builder[A, DoubleLinkedList[A]] =
+  @inline def newBuilder[A]: Builder[A, DoubleLinkedList[A]] =
