--- 2.12.15/scala/collection/mutable/LinkedList.scala
+++ old-collections/scala/collection/mutable/LinkedList.scala
@@ -117,9 +117,14 @@
  */
 @deprecated("low-level linked lists are deprecated", "2.11.0")
 object LinkedList extends SeqFactory[LinkedList] {
-  override def empty[A]: LinkedList[A] = new LinkedList[A]
-  implicit def canBuildFrom[A]: CanBuildFrom[Coll, A, LinkedList[A]] = ReusableCBF.asInstanceOf[GenericCanBuildFrom[A]]
+  @inline override def ReusableCBF: GenericCanBuildFrom[Nothing] = ReusableCBFInstance
+  private object ReusableCBFInstance extends GenericCanBuildFrom[Nothing] {
+    @inline override def apply() = newBuilder[Nothing]
+  }
 
-  def newBuilder[A]: Builder[A, LinkedList[A]] =
+  @inline override def empty[A]: LinkedList[A] = new LinkedList[A]
+  @inline implicit def canBuildFrom[A]: CanBuildFrom[Coll, A, LinkedList[A]] = ReusableCBF.asInstanceOf[GenericCanBuildFrom[A]]
+
+  @inline def newBuilder[A]: Builder[A, LinkedList[A]] =
     (new MutableList) mapResult ((l: MutableList[A]) => l.toLinkedList)
 }
