--- 2.13.6/scala/reflect/Manifest.scala
+++ overrides-2.13/scala/reflect/Manifest.scala
@@ -82,22 +82,22 @@
   def valueManifests: List[AnyValManifest[_]] =
     ManifestFactory.valueManifests
 
-  val Byte: ManifestFactory.ByteManifest = ManifestFactory.Byte
-  val Short: ManifestFactory.ShortManifest = ManifestFactory.Short
-  val Char: ManifestFactory.CharManifest = ManifestFactory.Char
-  val Int: ManifestFactory.IntManifest = ManifestFactory.Int
-  val Long: ManifestFactory.LongManifest = ManifestFactory.Long
-  val Float: ManifestFactory.FloatManifest = ManifestFactory.Float
-  val Double: ManifestFactory.DoubleManifest = ManifestFactory.Double
-  val Boolean: ManifestFactory.BooleanManifest = ManifestFactory.Boolean
-  val Unit: ManifestFactory.UnitManifest = ManifestFactory.Unit
+  @inline def Byte: ManifestFactory.ByteManifest = ManifestFactory.Byte
+  @inline def Short: ManifestFactory.ShortManifest = ManifestFactory.Short
+  @inline def Char: ManifestFactory.CharManifest = ManifestFactory.Char
+  @inline def Int: ManifestFactory.IntManifest = ManifestFactory.Int
+  @inline def Long: ManifestFactory.LongManifest = ManifestFactory.Long
+  @inline def Float: ManifestFactory.FloatManifest = ManifestFactory.Float
+  @inline def Double: ManifestFactory.DoubleManifest = ManifestFactory.Double
+  @inline def Boolean: ManifestFactory.BooleanManifest = ManifestFactory.Boolean
+  @inline def Unit: ManifestFactory.UnitManifest = ManifestFactory.Unit
 
-  val Any: Manifest[scala.Any] = ManifestFactory.Any
-  val Object: Manifest[java.lang.Object] = ManifestFactory.Object
-  val AnyRef: Manifest[scala.AnyRef] = ManifestFactory.AnyRef
-  val AnyVal: Manifest[scala.AnyVal] = ManifestFactory.AnyVal
-  val Null: Manifest[scala.Null] = ManifestFactory.Null
-  val Nothing: Manifest[scala.Nothing] = ManifestFactory.Nothing
+  @inline def Any: Manifest[scala.Any] = ManifestFactory.Any
+  @inline def Object: Manifest[java.lang.Object] = ManifestFactory.Object
+  @inline def AnyRef: Manifest[scala.AnyRef] = ManifestFactory.AnyRef
+  @inline def AnyVal: Manifest[scala.AnyVal] = ManifestFactory.AnyVal
+  @inline def Null: Manifest[scala.Null] = ManifestFactory.Null
+  @inline def Nothing: Manifest[scala.Nothing] = ManifestFactory.Nothing
 
   /** Manifest for the singleton type `value.type`. */
   def singleType[T <: AnyRef](value: AnyRef): Manifest[T] =
