--- 2.13.4/scala/Predef.scala
+++ overrides-2.13.4/scala/Predef.scala
@@ -163,1 +163,1 @@
-  val Map         = immutable.Map
+  @inline def Map = immutable.Map
@@ -165,1 +165,1 @@
-  val Set         = immutable.Set
+  @inline def Set = immutable.Set
@@ -167,2 +167,1 @@
-  /**
-   * Allows destructuring tuples with the same syntax as constructing them.
+  /**   * Allows destructuring tuples with the same syntax as constructing them.
@@ -172,2 +171,1 @@
-   *
-   * val c = tup match {
+   *   * val c = tup match {
@@ -177,2 +175,1 @@
-   * @group aliases
-   */
+   * @group aliases   */
@@ -182,2 +179,1 @@
-  // TODO undeprecated until Scala reflection becomes non-experimental
-  // @deprecated("this notion doesn't have a corresponding concept in 2.10, because scala.reflect.runtime.universe.TypeTag can capture arbitrary types. Use type tags instead of manifests, and there will be no need in opt manifests.", "2.10.0")
+  // TODO undeprecated until Scala reflection becomes non-experimental  // @deprecated("this notion doesn't have a corresponding concept in 2.10, because scala.reflect.runtime.universe.TypeTag can capture arbitrary types. Use type tags instead of manifests, and there will be no need in opt manifests.", "2.10.0")
@@ -191,1 +187,1 @@
-  val Manifest          = scala.reflect.Manifest
+  @inline def Manifest  = scala.reflect.Manifest
@@ -194,1 +190,1 @@
-  val NoManifest        = scala.reflect.NoManifest
+  @inline def NoManifest = scala.reflect.NoManifest
@@ -196,2 +192,1 @@
-  // TODO undeprecated until Scala reflection becomes non-experimental
-  // @deprecated("use scala.reflect.classTag[T] and scala.reflect.runtime.universe.typeTag[T] instead", "2.10.0")
+  // TODO undeprecated until Scala reflection becomes non-experimental  // @deprecated("use scala.reflect.classTag[T] and scala.reflect.runtime.universe.typeTag[T] instead", "2.10.0")
@@ -202,1 +197,0 @@
-
@@ -206,2 +200,1 @@
-   * A method that returns its input value.
-   * @tparam A type of the input value x.
+   * A method that returns its input value.   * @tparam A type of the input value x.
@@ -212,1 +205,0 @@
-
@@ -216,2 +208,1 @@
-   *  @return the implicit value of type `T`
-   *  @group utilities
+   *  @return the implicit value of type `T`   *  @group utilities
@@ -221,2 +212,1 @@
-  /** Used to mark code blocks as being expressions, instead of being taken as part of anonymous classes and the like.
-   *  This is just a different name for [[identity]].
+  /** Used to mark code blocks as being expressions, instead of being taken as part of anonymous classes and the like.   *  This is just a different name for [[identity]].
@@ -226,2 +216,1 @@
-   *             val x = new AnyRef
-   *             {
+   *             val x = new AnyRef   *             {
