--- 2.11.12/scala/reflect/ClassTag.scala
+++ overrides-2.11/scala/reflect/ClassTag.scala
@@ -115,27 +115,23 @@
  * Class tags corresponding to primitive types and constructor/extractor for ClassTags.
  */
 object ClassTag {
-  private val ObjectTYPE = classOf[java.lang.Object]
-  private val NothingTYPE = classOf[scala.runtime.Nothing$]
-  private val NullTYPE = classOf[scala.runtime.Null$]
+  @inline def Byte    : ClassTag[scala.Byte]       = Manifest.Byte
+  @inline def Short   : ClassTag[scala.Short]      = Manifest.Short
+  @inline def Char    : ClassTag[scala.Char]       = Manifest.Char
+  @inline def Int     : ClassTag[scala.Int]        = Manifest.Int
+  @inline def Long    : ClassTag[scala.Long]       = Manifest.Long
+  @inline def Float   : ClassTag[scala.Float]      = Manifest.Float
+  @inline def Double  : ClassTag[scala.Double]     = Manifest.Double
+  @inline def Boolean : ClassTag[scala.Boolean]    = Manifest.Boolean
+  @inline def Unit    : ClassTag[scala.Unit]       = Manifest.Unit
+  @inline def Any     : ClassTag[scala.Any]        = Manifest.Any
+  @inline def Object  : ClassTag[java.lang.Object] = Manifest.Object
+  @inline def AnyVal  : ClassTag[scala.AnyVal]     = Manifest.AnyVal
+  @inline def AnyRef  : ClassTag[scala.AnyRef]     = Manifest.AnyRef
+  @inline def Nothing : ClassTag[scala.Nothing]    = Manifest.Nothing
+  @inline def Null    : ClassTag[scala.Null]       = Manifest.Null
 
-  val Byte    : ClassTag[scala.Byte]       = Manifest.Byte
-  val Short   : ClassTag[scala.Short]      = Manifest.Short
-  val Char    : ClassTag[scala.Char]       = Manifest.Char
-  val Int     : ClassTag[scala.Int]        = Manifest.Int
-  val Long    : ClassTag[scala.Long]       = Manifest.Long
-  val Float   : ClassTag[scala.Float]      = Manifest.Float
-  val Double  : ClassTag[scala.Double]     = Manifest.Double
-  val Boolean : ClassTag[scala.Boolean]    = Manifest.Boolean
-  val Unit    : ClassTag[scala.Unit]       = Manifest.Unit
-  val Any     : ClassTag[scala.Any]        = Manifest.Any
-  val Object  : ClassTag[java.lang.Object] = Manifest.Object
-  val AnyVal  : ClassTag[scala.AnyVal]     = Manifest.AnyVal
-  val AnyRef  : ClassTag[scala.AnyRef]     = Manifest.AnyRef
-  val Nothing : ClassTag[scala.Nothing]    = Manifest.Nothing
-  val Null    : ClassTag[scala.Null]       = Manifest.Null
-
-  def apply[T](runtimeClass1: jClass[_]): ClassTag[T] =
+  @inline def apply[T](runtimeClass1: jClass[_]): ClassTag[T] =
     runtimeClass1 match {
       case java.lang.Byte.TYPE      => ClassTag.Byte.asInstanceOf[ClassTag[T]]
       case java.lang.Short.TYPE     => ClassTag.Short.asInstanceOf[ClassTag[T]]
@@ -146,11 +142,17 @@
       case java.lang.Double.TYPE    => ClassTag.Double.asInstanceOf[ClassTag[T]]
       case java.lang.Boolean.TYPE   => ClassTag.Boolean.asInstanceOf[ClassTag[T]]
       case java.lang.Void.TYPE      => ClassTag.Unit.asInstanceOf[ClassTag[T]]
-      case ObjectTYPE               => ClassTag.Object.asInstanceOf[ClassTag[T]]
-      case NothingTYPE              => ClassTag.Nothing.asInstanceOf[ClassTag[T]]
-      case NullTYPE                 => ClassTag.Null.asInstanceOf[ClassTag[T]]
-      case _                        => new ClassTag[T]{ def runtimeClass = runtimeClass1 }
+      case _                        =>
+        if (classOf[java.lang.Object] == runtimeClass1) {
+          ClassTag.Object.asInstanceOf[ClassTag[T]]
+        } else if (classOf[scala.runtime.Nothing$] == runtimeClass1) {
+          ClassTag.Nothing.asInstanceOf[ClassTag[T]]
+        } else if (classOf[scala.runtime.Null$] == runtimeClass1) {
+          ClassTag.Null.asInstanceOf[ClassTag[T]]
+        } else {
+          new ClassTag[T]{ def runtimeClass = runtimeClass1 }
+        }
     }
 
-  def unapply[T](ctag: ClassTag[T]): Option[Class[_]] = Some(ctag.runtimeClass)
+  @inline def unapply[T](ctag: ClassTag[T]): Option[Class[_]] = Some(ctag.runtimeClass)
 }
