--- scalalib/target/scalaSources/2.11.12/scala/collection/GenSet.scala	2017-11-03 00:26:48.000000000 +0100
+++ scalalib/overrides-2.11/scala/collection/GenSet.scala	2021-02-20 13:54:41.611200399 +0100
@@ -34,4 +34,9 @@
 object GenSet extends GenTraversableFactory[GenSet] {
-  implicit def canBuildFrom[A] = ReusableCBF.asInstanceOf[GenericCanBuildFrom[A]]
-  def newBuilder[A] = Set.newBuilder
+  @inline override def ReusableCBF: GenericCanBuildFrom[Nothing] = ReusableCBFInstance
+  private object ReusableCBFInstance extends GenericCanBuildFrom[Nothing] {
+    @inline override def apply() = newBuilder[Nothing]
+  }
+
+  @inline implicit def canBuildFrom[A] = ReusableCBF.asInstanceOf[GenericCanBuildFrom[A]]
+  @inline def newBuilder[A] = Set.newBuilder
 }
