--- scalalib/target/scalaSources/2.12.13/scala/collection/Seq.scala	2021-01-12 14:37:36.000000000 +0100
+++ scalalib/overrides-2.12/scala/collection/Seq.scala	2021-02-19 18:54:20.458914303 +0100
@@ -37,6 +37,10 @@
 object Seq extends SeqFactory[Seq] {
-  /** $genericCanBuildFromInfo */
-  implicit def canBuildFrom[A]: CanBuildFrom[Coll, A, Seq[A]] = ReusableCBF.asInstanceOf[GenericCanBuildFrom[A]]
+  @inline override def ReusableCBF: GenericCanBuildFrom[Nothing] = ReusableCBFInstance
+  private object ReusableCBFInstance extends GenericCanBuildFrom[Nothing] {
+    @inline override def apply() = newBuilder[Nothing]
+  }
 
-  def newBuilder[A]: Builder[A, Seq[A]] = immutable.Seq.newBuilder[A]
+  /** $genericCanBuildFromInfo */
+  @inline implicit def canBuildFrom[A]: CanBuildFrom[Coll, A, Seq[A]] = ReusableCBF.asInstanceOf[GenericCanBuildFrom[A]]
+  @inline def newBuilder[A]: Builder[A, Seq[A]] = immutable.Seq.newBuilder[A]
 }
