--- scalalib/target/scalaSources/2.12.13/scala/collection/Traversable.scala	2021-01-12 14:37:36.000000000 +0100
+++ scalalib/overrides-2.12/scala/collection/Traversable.scala	2021-02-19 18:54:20.482914339 +0100
@@ -96,2 +96,6 @@
 object Traversable extends TraversableFactory[Traversable] { self =>
+  @inline override def ReusableCBF: GenericCanBuildFrom[Nothing] = ReusableCBFInstance
+  private object ReusableCBFInstance extends GenericCanBuildFrom[Nothing] {
+    @inline override def apply() = newBuilder[Nothing]
+  }
 
@@ -101,5 +105,4 @@
   /** $genericCanBuildFromInfo */
-  implicit def canBuildFrom[A]: CanBuildFrom[Coll, A, Traversable[A]] = ReusableCBF.asInstanceOf[GenericCanBuildFrom[A]]
-
-  def newBuilder[A]: Builder[A, Traversable[A]] = immutable.Traversable.newBuilder[A]
+  @inline implicit def canBuildFrom[A]: CanBuildFrom[Coll, A, Traversable[A]] = ReusableCBF.asInstanceOf[GenericCanBuildFrom[A]]
+  @inline def newBuilder[A]: Builder[A, Traversable[A]] = immutable.Traversable.newBuilder[A]
 }
