--- scalalib/target/scalaSources/2.12.13/scala/collection/mutable/ArraySeq.scala	2021-01-12 14:37:36.000000000 +0100
+++ scalalib/overrides-2.12/scala/collection/mutable/ArraySeq.scala	2021-02-19 18:54:20.574914477 +0100
@@ -107,5 +107,10 @@
 object ArraySeq extends SeqFactory[ArraySeq] {
+  @inline override def ReusableCBF: GenericCanBuildFrom[Nothing] = ReusableCBFInstance
+  private object ReusableCBFInstance extends GenericCanBuildFrom[Nothing] {
+    @inline override def apply() = newBuilder[Nothing]
+  }
+
   /** $genericCanBuildFromInfo */
-  implicit def canBuildFrom[A]: CanBuildFrom[Coll, A, ArraySeq[A]] = ReusableCBF.asInstanceOf[GenericCanBuildFrom[A]]
-  def newBuilder[A]: Builder[A, ArraySeq[A]] =
+  @inline implicit def canBuildFrom[A]: CanBuildFrom[Coll, A, ArraySeq[A]] = ReusableCBF.asInstanceOf[GenericCanBuildFrom[A]]
+  @inline def newBuilder[A]: Builder[A, ArraySeq[A]] =
     new ArrayBuffer[A] mapResult { buf =>
