--- scalalib/target/scalaSources/2.12.13/scala/collection/mutable/LinkedList.scala	2021-01-12 14:37:36.000000000 +0100
+++ scalalib/overrides-2.12/scala/collection/mutable/LinkedList.scala	2021-02-19 18:54:20.578914483 +0100
@@ -119,6 +119,13 @@
 object LinkedList extends SeqFactory[LinkedList] {
-  override def empty[A]: LinkedList[A] = new LinkedList[A]
-  implicit def canBuildFrom[A]: CanBuildFrom[Coll, A, LinkedList[A]] = ReusableCBF.asInstanceOf[GenericCanBuildFrom[A]]
+  @inline override def ReusableCBF: GenericCanBuildFrom[Nothing] = ReusableCBFInstance
+  private object ReusableCBFInstance extends GenericCanBuildFrom[Nothing] {
+    @inline override def apply() = newBuilder[Nothing]
+  }
 
-  def newBuilder[A]: Builder[A, LinkedList[A]] =
+  @inline override def empty[A]: LinkedList[A] =
+    new LinkedList[A]
+
+  @inline implicit def canBuildFrom[A]: CanBuildFrom[Coll, A, LinkedList[A]] = ReusableCBF.asInstanceOf[GenericCanBuildFrom[A]]
+
+  @inline def newBuilder[A]: Builder[A, LinkedList[A]] =
     (new MutableList) mapResult ((l: MutableList[A]) => l.toLinkedList)
