--- 2.12.15/scala/collection/mutable/ResizableArray.scala
+++ old-collections/scala/collection/mutable/ResizableArray.scala
@@ -125,8 +125,12 @@
 }
 
 object ResizableArray extends SeqFactory[ResizableArray] {
-  implicit def canBuildFrom[A]: CanBuildFrom[Coll, A, ResizableArray[A]] =
+  @inline override def ReusableCBF: GenericCanBuildFrom[Nothing] = ReusableCBFInstance
+  private object ReusableCBFInstance extends GenericCanBuildFrom[Nothing] {
+    @inline override def apply() = newBuilder[Nothing]
+  }
+  @inline implicit def canBuildFrom[A]: CanBuildFrom[Coll, A, ResizableArray[A]] =
     ReusableCBF.asInstanceOf[GenericCanBuildFrom[A]]
 
-  def newBuilder[A]: Builder[A, ResizableArray[A]] = new ArrayBuffer[A]
+  @inline def newBuilder[A]: Builder[A, ResizableArray[A]] = new ArrayBuffer[A]
 }
