--- scalalib/target/scalaSources/2.12.13/scala/collection/mutable/Stack.scala	2021-01-12 14:37:36.000000000 +0100
+++ scalalib/overrides-2.12/scala/collection/mutable/Stack.scala	2021-02-19 18:54:20.582914489 +0100
@@ -28,2 +28,7 @@
 object Stack extends SeqFactory[Stack] {
+  @inline override def ReusableCBF: GenericCanBuildFrom[Nothing] = ReusableCBFInstance
+  private object ReusableCBFInstance extends GenericCanBuildFrom[Nothing] {
+    @inline override def apply() = newBuilder[Nothing]
+  }
+
   class StackBuilder[A] extends Builder[A, Stack[A]] {
@@ -35,5 +40,5 @@
 
-  implicit def canBuildFrom[A]: CanBuildFrom[Coll, A, Stack[A]] = ReusableCBF.asInstanceOf[GenericCanBuildFrom[A]]
-  def newBuilder[A]: Builder[A, Stack[A]] = new StackBuilder[A]
-  val empty: Stack[Nothing] = new Stack(Nil)
+  @inline implicit def canBuildFrom[A]: CanBuildFrom[Coll, A, Stack[A]] = ReusableCBF.asInstanceOf[GenericCanBuildFrom[A]]
+  @inline def newBuilder[A]: Builder[A, Stack[A]] = new StackBuilder[A]
+  @inline val empty: Stack[Nothing] = new Stack(Nil)
 }
