--- 2.13.12/scala/Array.scala
+++ overrides-2.13/scala/Array.scala
@@ -32,7 +32,7 @@
  *  where the array objects `a`, `b` and `c` have respectively the values
  *  `Array(1, 2)`, `Array(0, 0)` and `Array(1, 2, 0, 0)`.
  */
-object Array {
+private object EmptyArrays {
   val emptyBooleanArray = new Array[Boolean](0)
   val emptyByteArray    = new Array[Byte](0)
   val emptyCharArray    = new Array[Char](0)
@@ -42,7 +42,19 @@
   val emptyLongArray    = new Array[Long](0)
   val emptyShortArray   = new Array[Short](0)
   val emptyObjectArray  = new Array[Object](0)
+}
 
+object Array {
+  @inline def emptyBooleanArray = EmptyArrays.emptyBooleanArray
+  @inline def emptyByteArray    = EmptyArrays.emptyByteArray
+  @inline def emptyCharArray    = EmptyArrays.emptyCharArray
+  @inline def emptyDoubleArray  = EmptyArrays.emptyDoubleArray
+  @inline def emptyFloatArray   = EmptyArrays.emptyFloatArray
+  @inline def emptyIntArray     = EmptyArrays.emptyIntArray
+  @inline def emptyLongArray    = EmptyArrays.emptyLongArray
+  @inline def emptyShortArray   = EmptyArrays.emptyShortArray
+  @inline def emptyObjectArray  = EmptyArrays.emptyObjectArray
+
   /** Provides an implicit conversion from the Array object to a collection Factory */
   implicit def toFactory[A : ClassTag](dummy: Array.type): Factory[A, Array[A]] = new ArrayFactory(dummy)
   @SerialVersionUID(3L)
@@ -122,7 +134,8 @@
     * @see `java.util.Arrays#copyOf`
     */
   def copyOf[A](original: Array[A], newLength: Int): Array[A] = ((original: @unchecked) match {
-    case x: Array[BoxedUnit]  => newUnitArray(newLength).asInstanceOf[Array[A]]
+//  We cannot distinguish Array[BoxedUnit] from Array[Object] in Scala Native
+//	case x: Array[BoxedUnit]  => newUnitArray(newLength).asInstanceOf[Array[A]]
     case x: Array[AnyRef]     => java.util.Arrays.copyOf(x, newLength)
     case x: Array[Int]        => java.util.Arrays.copyOf(x, newLength)
     case x: Array[Double]     => java.util.Arrays.copyOf(x, newLength)
@@ -183,16 +196,7 @@
   // Subject to a compiler optimization in Cleanup.
   // Array(e0, ..., en) is translated to { val a = new Array(3); a(i) = ei; a }
   def apply[T: ClassTag](xs: T*): Array[T] = {
-    val len = xs.length
-    xs match {
-      case wa: immutable.ArraySeq[_] if wa.unsafeArray.getClass.getComponentType == classTag[T].runtimeClass =>
-        // We get here in test/files/run/sd760a.scala, `Array[T](t)` for
-        // a specialized type parameter `T`. While we still pay for two
-        // copies of the array it is better than before when we also boxed
-        // each element when populating the result.
-        ScalaRunTime.array_clone(wa.unsafeArray).asInstanceOf[Array[T]]
-      case _ =>
-        val array = new Array[T](len)
+    val array = new Array[T](xs.length)
         val iterator = xs.iterator
         var i = 0
         while (iterator.hasNext) {
@@ -200,7 +204,6 @@
         }
         array
     }
-  }
 
   /** Creates an array of `Boolean` objects */
   // Subject to a compiler optimization in Cleanup, see above.
@@ -577,7 +580,7 @@
   def unapplySeq[T](x: Array[T]): UnapplySeqWrapper[T] = new UnapplySeqWrapper(x)
 
   final class UnapplySeqWrapper[T](private val a: Array[T]) extends AnyVal {
-    def isEmpty: false = false
+    def isEmpty: Boolean = false
     def get: UnapplySeqWrapper[T] = this
     def lengthCompare(len: Int): Int = a.lengthCompare(len)
     def apply(i: Int): T = a(i)
