--- 2.12.17/scala/reflect/Manifest.scala
+++ overrides-2.12/scala/reflect/Manifest.scala
@@ -1,3 +1,4 @@
+
 /*
  * Scala (https://www.scala-lang.org)
  *
@@ -76,8 +77,8 @@
     case _                    => false
   }
   override def equals(that: Any): Boolean = this eq that.asInstanceOf[AnyRef]
-  @transient
-  override val hashCode = System.identityHashCode(this)
+  @transient @inline
+  override def hashCode = System.identityHashCode(this)
 }
 
 /** `ManifestFactory` defines factory methods for manifests.
@@ -241,9 +242,9 @@
   }
   val Unit: AnyValManifest[Unit] = new UnitManifest
 
-  private val ObjectTYPE = classOf[java.lang.Object]
-  private val NothingTYPE = classOf[scala.runtime.Nothing$]
-  private val NullTYPE = classOf[scala.runtime.Null$]
+  @inline private def ObjectTYPE = classOf[java.lang.Object]
+  @inline private def NothingTYPE = classOf[scala.runtime.Nothing$]
+  @inline private def NullTYPE = classOf[scala.runtime.Null$]
 
   @SerialVersionUID(1L)
   private class AnyManifest extends PhantomManifest[scala.Any](ObjectTYPE, "Any") {
@@ -323,8 +324,8 @@
   private abstract class PhantomManifest[T](_runtimeClass: Predef.Class[_],
                                             override val toString: String) extends ClassTypeManifest[T](None, _runtimeClass, Nil) {
     override def equals(that: Any): Boolean = this eq that.asInstanceOf[AnyRef]
-    @transient
-    override val hashCode = System.identityHashCode(this)
+    @transient @inline
+    override def hashCode = System.identityHashCode(this)
   }
 
   /** Manifest for the class type `clazz[args]`, where `clazz` is
