--- 2.12.15/scala/reflect/Manifest.scala
+++ overrides-2.12/scala/reflect/Manifest.scala
@@ -77,6 +77,6 @@
   }
   override def equals(that: Any): Boolean = this eq that.asInstanceOf[AnyRef]
-  @transient
-  override val hashCode = System.identityHashCode(this)
+  @transient @inline
+  override def hashCode = System.identityHashCode(this)
 }
 
@@ -106,5 +106,5 @@
     private def readResolve(): Any = Manifest.Byte
   }
-  val Byte: AnyValManifest[Byte] = new ByteManifest
+  @inline def Byte: AnyValManifest[Byte] = new ByteManifest
 
   @SerialVersionUID(1L)
@@ -122,5 +122,5 @@
     private def readResolve(): Any = Manifest.Short
   }
-  val Short: AnyValManifest[Short] = new ShortManifest
+  @inline def Short: AnyValManifest[Short] = new ShortManifest
 
   @SerialVersionUID(1L)
@@ -138,5 +138,5 @@
     private def readResolve(): Any = Manifest.Char
   }
-  val Char: AnyValManifest[Char] = new CharManifest
+  @inline def Char: AnyValManifest[Char] = new CharManifest
 
   @SerialVersionUID(1L)
@@ -154,5 +154,5 @@
     private def readResolve(): Any = Manifest.Int
   }
-  val Int: AnyValManifest[Int] = new IntManifest
+  @inline def Int: AnyValManifest[Int] = new IntManifest
 
   @SerialVersionUID(1L)
@@ -170,5 +170,5 @@
     private def readResolve(): Any = Manifest.Long
   }
-  val Long: AnyValManifest[Long] = new LongManifest
+  @inline def Long: AnyValManifest[Long] = new LongManifest
 
   @SerialVersionUID(1L)
@@ -186,5 +186,5 @@
     private def readResolve(): Any = Manifest.Float
   }
-  val Float: AnyValManifest[Float] = new FloatManifest
+  @inline def Float: AnyValManifest[Float] = new FloatManifest
 
   @SerialVersionUID(1L)
@@ -205,5 +205,5 @@
     private def readResolve(): Any = Manifest.Double
   }
-  val Double: AnyValManifest[Double] = new DoubleManifest
+  @inline def Double: AnyValManifest[Double] = new DoubleManifest
 
   @SerialVersionUID(1L)
@@ -221,5 +221,5 @@
     private def readResolve(): Any = Manifest.Boolean
   }
-  val Boolean: AnyValManifest[Boolean] = new BooleanManifest
+  @inline def Boolean: AnyValManifest[Boolean] = new BooleanManifest
 
   @SerialVersionUID(1L)
@@ -240,5 +240,5 @@
     private def readResolve(): Any = Manifest.Unit
   }
-  val Unit: AnyValManifest[Unit] = new UnitManifest
+  @inline def Unit: AnyValManifest[Unit] = new UnitManifest
 
   private val ObjectTYPE = classOf[java.lang.Object]
@@ -252,5 +252,5 @@
     private def readResolve(): Any = Manifest.Any
   }
-  val Any: Manifest[scala.Any] = new AnyManifest
+  @inline def Any: Manifest[scala.Any] = new AnyManifest
 
   @SerialVersionUID(1L)
@@ -260,7 +260,7 @@
     private def readResolve(): Any = Manifest.Object
   }
-  val Object: Manifest[java.lang.Object] = new ObjectManifest
+  @inline def Object: Manifest[java.lang.Object] = new ObjectManifest
 
-  val AnyRef: Manifest[scala.AnyRef] = Object.asInstanceOf[Manifest[scala.AnyRef]]
+  @inline def AnyRef: Manifest[scala.AnyRef] = Object.asInstanceOf[Manifest[scala.AnyRef]]
 
   @SerialVersionUID(1L)
@@ -270,5 +270,5 @@
     private def readResolve(): Any = Manifest.AnyVal
   }
-  val AnyVal: Manifest[scala.AnyVal] = new AnyValPhantomManifest
+  @inline def AnyVal: Manifest[scala.AnyVal] = new AnyValPhantomManifest
 
   @SerialVersionUID(1L)
@@ -279,5 +279,5 @@
     private def readResolve(): Any = Manifest.Null
   }
-  val Null: Manifest[scala.Null] = new NullManifest
+  @inline def Null: Manifest[scala.Null] = new NullManifest
 
   @SerialVersionUID(1L)
@@ -287,5 +287,5 @@
     private def readResolve(): Any = Manifest.Nothing
   }
-  val Nothing: Manifest[scala.Nothing] = new NothingManifest
+  @inline def Nothing: Manifest[scala.Nothing] = new NothingManifest
 
   @SerialVersionUID(1L)
@@ -324,6 +324,6 @@
                                             override val toString: String) extends ClassTypeManifest[T](None, _runtimeClass, Nil) {
     override def equals(that: Any): Boolean = this eq that.asInstanceOf[AnyRef]
-    @transient
-    override val hashCode = System.identityHashCode(this)
+    @transient @inline
+    override def hashCode = System.identityHashCode(this)
   }
 
