--- 2.12.14/scala/collection/generic/GenTraversableFactory.scala
+++ old-collections/scala/collection/generic/GenTraversableFactory.scala
@@ -44,2 +44,2 @@
-  private[this] val ReusableCBFInstance: GenericCanBuildFrom[Nothing] = new GenericCanBuildFrom[Nothing] {
-    override def apply() = newBuilder[Nothing]
+  private object ReusableCBFInstance extends GenericCanBuildFrom[Nothing] {
+    @inline override def apply() = newBuilder[Nothing]
@@ -47,1 +47,0 @@
-  def ReusableCBF: GenericCanBuildFrom[Nothing] = ReusableCBFInstance
@@ -49,0 +48,2 @@
+  @inline def ReusableCBF: GenericCanBuildFrom[Nothing] = ReusableCBFInstance
+
