--- scalalib/target/scalaSources/2.11.12/scala/collection/generic/GenTraversableFactory.scala	2017-11-03 00:26:48.000000000 +0100
+++ scalalib/overrides-2.11/scala/collection/generic/GenTraversableFactory.scala	2021-02-20 13:54:41.607200421 +0100
@@ -41,6 +41,7 @@
 
-  private[this] val ReusableCBFInstance: GenericCanBuildFrom[Nothing] = new GenericCanBuildFrom[Nothing] {
-    override def apply() = newBuilder[Nothing]
+  private object ReusableCBFInstance extends GenericCanBuildFrom[Nothing] {
+    @inline override def apply() = newBuilder[Nothing]
   }
-  def ReusableCBF: GenericCanBuildFrom[Nothing] = ReusableCBFInstance
+
+  @inline def ReusableCBF: GenericCanBuildFrom[Nothing] = ReusableCBFInstance
 
