--- scalalib/target/scalaSources/2.11.12/scala/collection/immutable/Traversable.scala	2017-11-03 00:26:48.000000000 +0100
+++ scalalib/overrides-2.11/scala/collection/immutable/Traversable.scala	2021-02-20 13:54:41.527200865 +0100
@@ -36,4 +36,9 @@
 object Traversable extends TraversableFactory[Traversable] {
-  implicit def canBuildFrom[A]: CanBuildFrom[Coll, A, Traversable[A]] = ReusableCBF.asInstanceOf[GenericCanBuildFrom[A]]
-  def newBuilder[A]: Builder[A, Traversable[A]] = new mutable.ListBuffer
+  @inline override def ReusableCBF: GenericCanBuildFrom[Nothing] = ReusableCBFInstance
+  private object ReusableCBFInstance extends GenericCanBuildFrom[Nothing] {
+    @inline override def apply() = newBuilder[Nothing]
+  }
+
+  @inline implicit def canBuildFrom[A]: CanBuildFrom[Coll, A, Traversable[A]] = ReusableCBF.asInstanceOf[GenericCanBuildFrom[A]]
+  @inline def newBuilder[A]: Builder[A, Traversable[A]] = new mutable.ListBuffer
 }
