--- scalalib/target/scalaSources/2.11.12/scala/compat/Platform.scala	2017-11-03 00:26:48.000000000 +0100
+++ scalalib/overrides-2.11/scala/compat/Platform.scala	2021-02-26 19:45:05.545638437 +0100
@@ -104,11 +104,9 @@
 
   /** The default line separator.
    *
-   * On the JVM, this is equivalent to calling the method:
-   * `System.getProperty("line.separator")`
-   * with a default value of "\n".
+   * On the Scala Native, this is always "\n".
    */
-  val EOL = scala.util.Properties.lineSeparator
+  val EOL = "\n"
 
   /** The current time in milliseconds. The time is counted since 1 January 1970
     * UTC.
