--- 2.11.12/scala/compat/Platform.scala
+++ overrides-2.11/scala/compat/Platform.scala
@@ -107,3 +107,1 @@
-   * On the JVM, this is equivalent to calling the method:
-   * `System.getProperty("line.separator")`
-   * with a default value of "\n".
+   * On the Scala Native, this is always "\n".
@@ -111,1 +109,1 @@
-  val EOL = scala.util.Properties.lineSeparator
+  val EOL = "\n"
@@ -113,2 +111,1 @@
-  /** The current time in milliseconds. The time is counted since 1 January 1970
-    * UTC.
+  /** The current time in milliseconds. The time is counted since 1 January 1970    * UTC.
@@ -118,2 +115,1 @@
-    */
-  @inline
+    */  @inline
@@ -123,2 +119,1 @@
-   *
-   * This is a request that the underlying JVM runs the garbage collector.
+   *   * This is a request that the underlying JVM runs the garbage collector.
@@ -128,2 +123,1 @@
-  @inline
-  def collectGarbage(): Unit = System.gc()
+  @inline  def collectGarbage(): Unit = System.gc()
@@ -133,2 +127,1 @@
-  def defaultCharsetName: String = java.nio.charset.Charset.defaultCharset.name
-}
+  def defaultCharsetName: String = java.nio.charset.Charset.defaultCharset.name}
