--- scalalib/target/scalaSources/2.12.13/scala/collection/immutable/IndexedSeq.scala	2021-01-12 14:37:36.000000000 +0100
+++ scalalib/overrides-2.12/scala/collection/immutable/IndexedSeq.scala	2021-02-19 18:54:20.590914501 +0100
@@ -45,9 +45,13 @@
   class Impl[A](buf: ArrayBuffer[A]) extends AbstractSeq[A] with IndexedSeq[A] with Serializable {
-    def length = buf.length
-    def apply(idx: Int) = buf.apply(idx)
+    @inline def length = buf.length
+    @inline def apply(idx: Int) = buf.apply(idx)
   }
-  def newBuilder[A]: Builder[A, IndexedSeq[A]] = Vector.newBuilder[A]
-
+  @inline def newBuilder[A]: Builder[A, IndexedSeq[A]] = Vector.newBuilder[A]
   implicit def canBuildFrom[A]: CanBuildFrom[Coll, A, IndexedSeq[A]] =
     ReusableCBF.asInstanceOf[GenericCanBuildFrom[A]]
+
+  @inline override def ReusableCBF: GenericCanBuildFrom[Nothing] = ReusableCBFInstance
+  private object ReusableCBFInstance extends GenericCanBuildFrom[Nothing] {
+    @inline override def apply() = newBuilder[Nothing]
+  }
 }
