--- 2.12.15/scala/collection/immutable/IndexedSeq.scala
+++ old-collections/scala/collection/immutable/IndexedSeq.scala
@@ -43,10 +43,14 @@
  */
 object IndexedSeq extends IndexedSeqFactory[IndexedSeq] {
   class Impl[A](buf: ArrayBuffer[A]) extends AbstractSeq[A] with IndexedSeq[A] with Serializable {
-    def length = buf.length
-    def apply(idx: Int) = buf.apply(idx)
+    @inline def length = buf.length
+    @inline def apply(idx: Int) = buf.apply(idx)
   }
-  def newBuilder[A]: Builder[A, IndexedSeq[A]] = Vector.newBuilder[A]
+  @inline override def ReusableCBF: GenericCanBuildFrom[Nothing] = ReusableCBFInstance
+  private object ReusableCBFInstance extends GenericCanBuildFrom[Nothing] {
+    @inline override def apply() = newBuilder[Nothing]
+  }
+  @inline def newBuilder[A]: Builder[A, IndexedSeq[A]] = Vector.newBuilder[A]
 
   implicit def canBuildFrom[A]: CanBuildFrom[Coll, A, IndexedSeq[A]] =
     ReusableCBF.asInstanceOf[GenericCanBuildFrom[A]]
