--- scalalib/target/scalaSources/2.12.13/scala/collection/immutable/Stream.scala	2021-01-12 14:37:36.000000000 +0100
+++ scalalib/overrides-2.12/scala/collection/immutable/Stream.scala	2021-02-19 18:54:20.610914530 +0100
@@ -1078,2 +1078,6 @@
 object Stream extends SeqFactory[Stream] {
+  @inline override def ReusableCBF: GenericCanBuildFrom[Nothing] = ReusableCBFInstance
+  private object ReusableCBFInstance extends GenericCanBuildFrom[Nothing] {
+    @inline override def apply() = newBuilder[Nothing]
+  }
 
@@ -1089,8 +1093,8 @@
 
-  implicit def canBuildFrom[A]: CanBuildFrom[Coll, A, Stream[A]] =
-    ReusableCBF.asInstanceOf[CanBuildFrom[Coll, A, Stream[A]]]
-  private[this] val ReusableCBF = new StreamCanBuildFrom[Any]
+  @inline implicit def canBuildFrom[A]: CanBuildFrom[Coll, A, Stream[A]] = new StreamCanBuildFrom[A]
 
   /** Creates a new builder for a stream */
-  def newBuilder[A]: Builder[A, Stream[A]] = new StreamBuilder[A]
+  @inline def newBuilder[A]: Builder[A, Stream[A]] = new StreamBuilder[A]
+
+  import scala.collection.{Iterable, Seq, IndexedSeq}
 
@@ -1113,6 +1117,6 @@
   /** The empty stream */
-  override def empty[A]: Stream[A] = Empty
+  @inline override def empty[A]: Stream[A] = Empty
 
   /** A stream consisting of given elements */
-  override def apply[A](xs: A*): Stream[A] = xs.toStream
+  @inline override def apply[A](xs: A*): Stream[A] = xs.toStream
 
