--- scalalib/target/scalaSources/2.12.13/scala/collection/immutable/Stream.scala	2021-01-12 14:37:36.000000000 +0100
+++ scalalib/overrides-2.12/scala/collection/immutable/Stream.scala	2021-02-26 16:43:39.134508255 +0100
@@ -1076,6 +1076,10 @@
  * @since   2.8
  */
 object Stream extends SeqFactory[Stream] {
+  @inline override def ReusableCBF: GenericCanBuildFrom[Nothing] = ReusableCBFInstance
+  private object ReusableCBFInstance extends GenericCanBuildFrom[Nothing] {
+    @inline override def apply() = newBuilder[Nothing]
+  }
 
   /** The factory for streams.
    *  @note Methods such as map/flatMap will not invoke the `Builder` factory,
@@ -1087,12 +1091,12 @@
    */
   class StreamCanBuildFrom[A] extends GenericCanBuildFrom[A]
 
-  implicit def canBuildFrom[A]: CanBuildFrom[Coll, A, Stream[A]] =
-    ReusableCBF.asInstanceOf[CanBuildFrom[Coll, A, Stream[A]]]
-  private[this] val ReusableCBF = new StreamCanBuildFrom[Any]
+  @inline implicit def canBuildFrom[A]: CanBuildFrom[Coll, A, Stream[A]] = new StreamCanBuildFrom[A]
 
   /** Creates a new builder for a stream */
-  def newBuilder[A]: Builder[A, Stream[A]] = new StreamBuilder[A]
+  @inline def newBuilder[A]: Builder[A, Stream[A]] = new StreamBuilder[A]
+
+  import scala.collection.{Iterable, Seq, IndexedSeq}
 
   /** A builder for streams
    *  @note This builder is lazy only in the sense that it does not go downs the spine
@@ -1111,10 +1115,10 @@
   }
 
   /** The empty stream */
-  override def empty[A]: Stream[A] = Empty
+  @inline override def empty[A]: Stream[A] = Empty
 
   /** A stream consisting of given elements */
-  override def apply[A](xs: A*): Stream[A] = xs.toStream
+  @inline override def apply[A](xs: A*): Stream[A] = xs.toStream
 
   /** A wrapper class that adds `#::` for cons and `#:::` for concat as operations
    *  to streams.
