--- scalalib/target/scalaSources/2.12.13/scala/collection/immutable/Vector.scala	2021-01-12 14:37:36.000000000 +0100
+++ scalalib/overrides-2.12/scala/collection/immutable/Vector.scala	2021-02-19 18:54:20.658914603 +0100
@@ -24,4 +24,9 @@
 object Vector extends IndexedSeqFactory[Vector] {
-  def newBuilder[A]: Builder[A, Vector[A]] = new VectorBuilder[A]
-  implicit def canBuildFrom[A]: CanBuildFrom[Coll, A, Vector[A]] =
+  @inline override def ReusableCBF: GenericCanBuildFrom[Nothing] = ReusableCBFInstance
+  private object ReusableCBFInstance extends GenericCanBuildFrom[Nothing] {
+    @inline override def apply() = newBuilder[Nothing]
+  }
+
+  @inline def newBuilder[A]: Builder[A, Vector[A]] = new VectorBuilder[A]
+  @inline implicit def canBuildFrom[A]: CanBuildFrom[Coll, A, Vector[A]] =
     ReusableCBF.asInstanceOf[GenericCanBuildFrom[A]]
