--- scalalib/target/scalaSources/2.11.12/scala/collection/mutable/ArraySeq.scala	2017-11-03 00:26:48.000000000 +0100
+++ scalalib/overrides-2.11/scala/collection/mutable/ArraySeq.scala	2021-02-20 13:54:41.707199865 +0100
@@ -106,5 +106,10 @@
 object ArraySeq extends SeqFactory[ArraySeq] {
+  @inline override def ReusableCBF: GenericCanBuildFrom[Nothing] = ReusableCBFInstance
+  private object ReusableCBFInstance extends GenericCanBuildFrom[Nothing] {
+    @inline override def apply() = newBuilder[Nothing]
+  }
+
   /** $genericCanBuildFromInfo */
-  implicit def canBuildFrom[A]: CanBuildFrom[Coll, A, ArraySeq[A]] = ReusableCBF.asInstanceOf[GenericCanBuildFrom[A]]
-  def newBuilder[A]: Builder[A, ArraySeq[A]] =
+  @inline implicit def canBuildFrom[A]: CanBuildFrom[Coll, A, ArraySeq[A]] = ReusableCBF.asInstanceOf[GenericCanBuildFrom[A]]
+  @inline def newBuilder[A]: Builder[A, ArraySeq[A]] =
     new ArrayBuffer[A] mapResult { buf =>
