--- 2.11.12/scala/collection/mutable/DoubleLinkedList.scala
+++ overrides-2.11/scala/collection/mutable/DoubleLinkedList.scala
@@ -81,9 +81,9 @@
 @deprecated("Low-level linked lists are deprecated.", "2.11.0")
 object DoubleLinkedList extends SeqFactory[DoubleLinkedList] {
   /** $genericCanBuildFromInfo */
-  implicit def canBuildFrom[A]: CanBuildFrom[Coll, A, DoubleLinkedList[A]] = ReusableCBF.asInstanceOf[GenericCanBuildFrom[A]]
+  @inline implicit def canBuildFrom[A]: CanBuildFrom[Coll, A, DoubleLinkedList[A]] = ReusableCBF.asInstanceOf[GenericCanBuildFrom[A]]
 
-  def newBuilder[A]: Builder[A, DoubleLinkedList[A]] =
+  @inline def newBuilder[A]: Builder[A, DoubleLinkedList[A]] =
     new Builder[A, DoubleLinkedList[A]] {
       def emptyList() = new DoubleLinkedList[A]()
       var current = emptyList()
