--- scalalib/target/scalaSources/2.11.12/scala/collection/mutable/Seq.scala	2017-11-03 00:26:48.000000000 +0100
+++ scalalib/overrides-2.11/scala/collection/mutable/Seq.scala	2021-02-20 13:54:41.663200110 +0100
@@ -42,4 +42,12 @@
 object Seq extends SeqFactory[Seq] {
-  implicit def canBuildFrom[A]: CanBuildFrom[Coll, A, Seq[A]] = ReusableCBF.asInstanceOf[GenericCanBuildFrom[A]]
-  def newBuilder[A]: Builder[A, Seq[A]] = new ArrayBuffer
+  @inline override def ReusableCBF: GenericCanBuildFrom[Nothing] = ReusableCBFInstance
+  private object ReusableCBFInstance extends GenericCanBuildFrom[Nothing] {
+    @inline override def apply() = newBuilder[Nothing]
+  }
+
+  @inline implicit def canBuildFrom[A]: CanBuildFrom[Coll, A, Seq[A]] =
+    ReusableCBF.asInstanceOf[GenericCanBuildFrom[A]]
+
+  @inline def newBuilder[A]: Builder[A, Seq[A]] =
+    new ArrayBuffer
 }
