--- scalalib/target/scalaSources/2.11.12/scala/collection/mutable/Traversable.scala	2017-11-03 00:26:48.000000000 +0100
+++ scalalib/overrides-2.11/scala/collection/mutable/Traversable.scala	2021-02-20 13:54:41.795199377 +0100
@@ -35,4 +35,12 @@
 object Traversable extends TraversableFactory[Traversable] {
-  implicit def canBuildFrom[A]: CanBuildFrom[Coll, A, Traversable[A]] = ReusableCBF.asInstanceOf[GenericCanBuildFrom[A]]
-  def newBuilder[A]: Builder[A, Traversable[A]] = new ArrayBuffer
+  @inline override def ReusableCBF: GenericCanBuildFrom[Nothing] = ReusableCBFInstance
+  private object ReusableCBFInstance extends GenericCanBuildFrom[Nothing] {
+    @inline override def apply() = newBuilder[Nothing]
+  }
+
+  @inline implicit def canBuildFrom[A]: CanBuildFrom[Coll, A, Traversable[A]] =
+    ReusableCBF.asInstanceOf[GenericCanBuildFrom[A]]
+
+  @inline def newBuilder[A]: Builder[A, Traversable[A]] =
+    new ArrayBuffer
 }
